### ISSUE TITLE
Test failure in monitor_test.py::test_cleanup_on_driver_exit_many_redis_shards.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1332]
I saw the following error. I first saw it in #1331.

```
======================================================================
FAIL: testCleanupOnDriverExitManyRedisShards (__main__.MonitorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/monitor_test.py", line 88, in testCleanupOnDriverExitManyRedisShards
    self._testCleanupOnDriverExit(num_redis_shards=5)
  File "test/monitor_test.py", line 79, in _testCleanupOnDriverExit
    self.assertEqual((0, 1), StateSummary()[:2])
AssertionError: Tuples differ: (0, 1) != (4, 3)

First differing element 0:
0
4

- (0, 1)
+ (4, 3)
```

cc @concretevitamin

**Comment by robertnishihara** [comment]
This error is still present, just saw it in #1831.

**Comment by robertnishihara** [comment]
This test has been failing a lot recently

```
python -m pytest -v test/monitor_test.py::test_cleanup_on_driver_exit_many_redis_shards
```

**Comment by robertnishihara** [comment]
Fixed by #3840.
